### PR TITLE
Add support to GeoStore for 'intersects' logic

### PR DESCRIPTION
**Pull request by hogpilot**
Added a new method (`GeoStore.prototype.intersects`) to add searching the geostore for geometry intersection (in addition to `contains` and `within`).

### CODE REVIEW

File: spec/geostoreSpec.js [hunk 1]
```diff
@@ -506,7 +506,7 @@ describe("geostore", function() {
 
         gs.within({
           "type": "Polygon",
-          "coordinates": [ [ [ -122, 45, 0.0 ], [ -123, 45, 0.0 ], [ -123, 46, 0.0 ], [ -122, 46, 0.0 ], [ -122, 45, 0.0 ] ] ]
+          "coordinates": [ [ [ -121, 45, 0.0 ], [ -123, 45, 0.0 ], [ -123, 46, 0.0 ], [ -121, 46, 0.0 ], [ -121, 45, 0.0 ] ] ]
```
Comment (hogpilot) [inline spec/geostoreSpec.js]: Although this check has nothing to do with the new `intersects` logic, it was failing due to the result polygon not contained by the search polygon.

File: spec/geostoreSpec.js [hunk 2]
```diff
@@ -517,7 +517,7 @@ describe("geostore", function() {
         function (err, res) {
           expect(err).toEqual(null);
           expect(res.length).toEqual(1);
-          expect(res.id).toEqual(1);
+          expect(res[0].id).toEqual(1);
```
Comment (hogpilot) [inline spec/geostoreSpec.js]: Although this check has nothing to do with the new intersects logic, it was failing due to `res` being an Array vs checking the first item in `res`.

File: src/geostore.js [hunk 3]
```diff
@@ -427,6 +427,139 @@
 
   };
 
+  GeoStore.prototype.intersects = function(geojson){
+    var args = Array.prototype.slice.call(arguments);
```
Comment (hogpilot) [inline src/geostore.js]: This is almost an exact copy of the `GeoStore.prototype.contains` method, the only change being the name and the use of `intersects` on line 499.

**Comment by hogpilot** [comment]
This maybe should close #3

**Comment by hogpilot** [comment]
If/when this PR is closed, I'll also add a PR to [Esri/terraformer](https://github.com/Esri/terraformer) to update [one of the examples](http://terraformer.io/examples/browser/arcgis-geojson/index.html) to show this new capability.

**Comment by jgravois** [comment]
i'm glad to hear that introducing support for testing intersections was straightforward! i took a stab at it this afternoon myself hoping to come up with something a little more [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

https://github.com/Esri/terraformer-geostore/blob/0186bfd6739a59422cd14d8ecb50e25af9df96d5/src/geostore.js#L291-L313

how do you feel about doing it that way?

**Comment by hogpilot** [comment]
@jgravois I agree completely. I also expanded your example to include `within` (since it was also 90%+ similar to the other two).  See my latest commit.

**Comment by hogpilot** [comment]
I've noticed a few other places where we can improve the code by being more DRY, perhaps this justifies a new PR for that purpose?  I'll be glad to start one, if all agree.

**Comment by jgravois** [comment]
all tests are passing locally. thx @hogpilot!

**Pull request merged**